### PR TITLE
OCD-3894 - Combine Developer Details and Website Change Request to Developer Demographics Change Request

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-resources/src/main/resources/lookup.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 changerequest.website=1
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/lookup.properties]: Should this be removed?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/dao/ChangeRequestDeveloperDemographicsDAO.java [hunk 2]
```diff
@@ -82,24 +84,24 @@ private ChangeRequestDeveloperDetailsEntity getNewEntity(ChangeRequest cr, Chang
...
                 .getResultList();
 
         if (result == null || result.size() == 0) {
             throw new EntityRetrievalException(
-                    "Data error. Change request developer details not found in database.");
+                    "Data error. Change request developer demographics not found in database.");
         } else if (result.size() > 1) {
             throw new EntityRetrievalException(
-                    "Data error. Duplicate change request developer details in database.");
+                    "Data error. Duplicate change request developer demographic in database.");
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/dao/ChangeRequestDeveloperDemographicsDAO.java]: The error on line 101 has `demographics`. This one is singular
```suggestion
                    "Data error. Duplicate change request developer demographics in database.");
```

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/domain/ChangeRequestType.java [hunk 3]
```diff
@@ -17,7 +17,8 @@ public class ChangeRequestType implements Serializable {
     private Long id;
     private String name;
 
-    public Boolean isDemographic() {
+    //TODO - This needs to be changed
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/domain/ChangeRequestType.java]: Do we need to do this? Or is this okay for now?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java [hunk 4]
```diff
@@ -1,65 +1,53 @@
...
             } else {
                 boolean areDetailsValid = true;
-                HashMap<String, Object> crDetails = (HashMap) context.getNewChangeRequest().getDetails();
-                if (!isChangeRequestSelfDevloperValid(crDetails)) {
+                if (!isChangeRequestSelfDevloperValid(details)) {
                     getMessages().add(getErrorMessage("changeRequest.details.selfDeveloper.missing"));
                     areDetailsValid = false;
-                } else if (!isChangeRequestAddressValid(crDetails)) {
+                } else if (!isChangeRequestAddressValid(details)) {
                     getMessages().add(getErrorMessage("changeRequest.details.address.missing"));
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java]: These keys changed to `changeRequest.demographics.*`

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestDetailsUpdateValidation.java [hunk 5]
```diff
@@ -1,38 +1,34 @@
...
-                    return isChangeRequestWebsiteValid((HashMap) context.getNewChangeRequest().getDetails());
-                }
-            }
-        }
-
-        return true;
-    }
+//        if (context.getResourcePermissions().isUserRoleDeveloperAdmin()) {
+//            if (context.getNewChangeRequest().getDetails() != null) {
+//                if (isChangeRequestWebsite(context)) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestDetailsUpdateValidation.java]: There's a lot of commented out stuff here. What should be done with it?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/SelfDeveloperValidation.java [hunk 6]
```diff
@@ -1,27 +1,19 @@
...
 public class SelfDeveloperValidation extends ValidationRule<ChangeRequestValidationContext> {
 
     @Override
     public boolean isValid(ChangeRequestValidationContext context) {
         if (context.getResourcePermissions().isUserRoleDeveloperAdmin()) {
-            HashMap<String, Object> map = (HashMap) context.getNewChangeRequest().getDetails();
-            if (map.containsKey("selfDeveloper") && !isChangeRequestSelfDeveloperValid(map)) {
+            ChangeRequestDeveloperDemographics details = (ChangeRequestDeveloperDemographics) context.getNewChangeRequest().getDetails();
+            if (details.getSelfDeveloper() == null) {
                 getMessages().add(getErrorMessage("changeRequest.details.selfDeveloper.invalid"));
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/SelfDeveloperValidation.java]: Same error key issue as above. Don't want to miss it

**Review by andlar — CHANGES_REQUESTED** [review]
I see a change in the "CuresChartsOverTime.xlsx" file, but can't tell what it is. That worries me, as that file doesn't have anything to do with this ticket

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java [hunk 7]
```diff
@@ -1,35 +1,30 @@
...
+                    getMessages().add(getErrorMessage("changeRequest.demographics.address.missing"));
                     areDetailsValid = false;
-                } else if (!isChangeRequestAddressValid(crDetails)) {
-                    getMessages().add(getErrorMessage("changeRequest.details.address.missing"));
+                } else if (!isChangeRequestContactValid(details)) {
+                    getMessages().add(getErrorMessage("changeRequest.demographics.contact.missing"));
                     areDetailsValid = false;
-                } else if (!isChangeRequestContactValid(crDetails)) {
-                    getMessages().add(getErrorMessage("changeRequest.details.contact.missing"));
+                } else if (!isChangeRequestContactValid(details)) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java]: It looks like this "else if" is the same as the one above it. Am I reading this diff right?

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java (hunk 7 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java]: Good catch - it was a duplicate.

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**